runtime/interpreter/value.go

```diff
@@ -8179,6 +8180,17 @@ func NewPublicKeyValue(
 	// Validate the public key, and initialize 'isValid' field.
 	publicKeyValue.Fields.Set(sema.PublicKeyIsValidField, validationFunction(publicKeyValue))
 
+	// Public key value to string should include the key even though it is a computed field
+	stringerFields := NewStringValueOrderedMap()
+	stringerFields.Set(sema.PublicKeyPublicKeyField, publicKey.Copy())
+	publicKeyValue.Fields.Foreach(func(key string, value Value) {
+		stringerFields.Set(key, value)
+	})
+	stringer := func() string {
+		return formatComposite(string(publicKeyValue.TypeID()), stringerFields)
+	}
+	publicKeyValue.stringer = stringer
```

Comment:
Could we maybe delay the construction of the map required for the formatting until it is actually is needed? Also, I think the `ForEach` can be inlined. Maybe something like this:

```suggestion
	// Public key value to string should include the key even though it is a computed field
	var *stringerFields *StringValueOrderedMap
	publicKeyValue.stringer = func() string {
		if stringerFields == nil {
			stringerFields = NewStringValueOrderedMap()
			stringerFields.Set(sema.PublicKeyPublicKeyField, publicKey.Copy())
			publicKeyValue.Fields.Foreach(stringerFields.Set)
		}
		return formatComposite(string(publicKeyValue.TypeID()), stringerFields)
	}
```

Reply:
stringerFields.Set is a `func (key string, value Value) (oldValue Value, present bool)` Foreach expects `func (key string, value Value)` so inlining like that wont work

Reply:
Ah, right! 👍